docs/glyph-inspector.html

```diff
@@ -339,8 +384,117 @@ <h1>Free Software</h1>
     hline('Typo Descender', font.tables.os2.sTypoDescender);
 }
 
+function arrayBufferToBase64(buffer) {
+    const CHUNK_SIZE = 8192;
+    let result = '';
+    const uintArray = new Uint8Array(buffer);
+
+    for (let i = 0; i < uintArray.length; i += CHUNK_SIZE) {
+        const chunk = uintArray.subarray(i, i + CHUNK_SIZE);
+        result += String.fromCharCode.apply(null, chunk);
+    }
+
+    return btoa(result);
+}
+
+function base64ToArrayBuffer(base64) {
+    return Uint8Array.from(atob(base64), c => c.charCodeAt(0)).buffer;
+}
+
+window.redraw = function(withColors = true) {
+    if ( withColors ) {
+        updateColrOptions();
+    }
+    const selectedPage = document.querySelector('.page-selected');
+    const pageIndex = Array.from(selectedPage.parentElement.children).indexOf(selectedPage);    
+    displayGlyphPage(pageIndex);
+    const glyphIndex = document.querySelector('[data-glyph-index]')?.dataset.glyphIndex;
+    if(glyphIndex) {
+        displayGlyph(glyphIndex);
+        displayGlyphData(glyphIndex);
+    }
+}
+
+function throttleDebounce(func, wait) {
+    let timeout = null;
+    let lastFunc;
+    let lastRan;
+
+    return function() {
+        const context = this;
+        const args = arguments;
+        if (!lastRan) {
+            func.apply(context, args);
+            lastRan = Date.now();
+        } else {
+            clearTimeout(lastFunc);
+            lastFunc = setTimeout(function() {
+                if ((Date.now() - lastRan) >= wait) {
+                    func.apply(context, args);
+                    lastRan = Date.now();
+                }
+            }, Math.max(wait - (Date.now() - lastRan), 0));
+        }
+    };
+}
+
+function updateColrOptions() {
+    const palettes = window.font.palettes.getAll('hexa');
+    const palettesDiv = document.getElementById('palettes');
+    form.drawLayers.checked = window.fontOptions.drawLayers;
+    palettesDiv.innerHTML = '';
+    let markup = '';
+    if ( palettes.length ) {
+        palettes.forEach((palette, idx) => {
+            markup += `<dt><label><input type="radio" name="usePalette[]" oninput="window.fontOptions.usePalette = parseInt(this.value); window.redraw();" value="${idx}" ${idx === window.fontOptions.usePalette ? ' checked' : ''}><strong>↪ Palette ${idx}</strong></label> <button type="button" class="color-swatch" onclick="window.font.palettes.delete(${idx}); redraw();">❌</button></dt><dd><div class="color-swatches">`;
```

Comment:
At this point we shall move to a `Element` builder like
```js
const el = (tag, props={}, ch=[]) => ch.reduce((e,c) => (e.appendChild(c),e),Object.assign(document.createElement(tag),props))
```

so this long HTML string would become this (also long) declaration
```js
el("dt",{},[
    el("label",{},[
        el("input",{ type:"radio", name:"usePalette[]", value:idx,
            checked:idx === fontOptions.usePalette,
            oninput(){
                fontOptions.usePalette = parseInt(this.value);
                redraw();
            }
        }),
        el("strong",{innerText:`↪ Palette ${idx}`})
    ]),
    el("button", {innerText:"❌",type:"button",classList:"color-swatch", onclick(){
        font.palettes.delete(idx);
        redraw();
    }})
])
```

This prevent any HTML syntax error (which is not obvious to see in string form) Also, this prevent any HTML injection

Reply:
Sure, let's do that in the future. We should also get rid of the code duplication across the different html files' script blocks, by moving the code (or at least the duplicated code) to a common file, just like the css. But I don't want to bloat this PR more than it already is.